### DISCUSSION TITLE
Could I combine the LiquidMenu with the LcdProgressBar library?

### DISCUSSION THREAD

**Discussion by GstamelosJ** [VasilKalchev/LiquidMenu#84] [Q&A]
I'm working on a project for measuring a tank level and I haven't found any way to project the progressbar on the LCD screen.
I modified also the LcdProgressBar library changing the void LcdProgressBar::Draw(value) method to char* LcdProgressBar::Draw(value) for returning the string that represents the progress value but I didn't manage to fix it.
Any idea?

**Comment by VasilKalchev** [comment]
Hello, @GstamelosJ and sorry for the late reply.

I'm guessing that in the method `LcdProgressBar::Draw(value)` you've replaced
```
_lcd->setCursor(0, _row);
_lcd->print(progressBarString);
```
with a return to the address of `char progressBarString[_numCols + 1];`, something like:
```
return progressBarString; // make sure you are returning the address to the array
```
and that you've made the string static:
```
`static char progressBarString[_numCols + 1];`
```
otherwise its place in memory will be deallocated when it goes out of scope and it could be overwritten. Another option is to declare the string in global scope.

Other potential problems are with the unusual way that the "LiquidMenu" library handles dynamic text. Take a look at the example ["C_functions_menu.ino"](https://github.com/VaSe7u/LiquidMenu/blob/432c09e1d5cacdfe62fb35d7f4e3180302e5f66c/examples/C_functions_menu/C_functions_menu.ino), it first creates a char array [`char isFading_text[4];`](https://github.com/VaSe7u/LiquidMenu/blob/432c09e1d5cacdfe62fb35d7f4e3180302e5f66c/examples/C_functions_menu/C_functions_menu.ino#L109), attaches it to a `LiquidLine` object and uses [`strncpy(isFading_text, string_off, sizeof(string_off));`](https://github.com/VaSe7u/LiquidMenu/blob/432c09e1d5cacdfe62fb35d7f4e3180302e5f66c/examples/C_functions_menu/C_functions_menu.ino#L163) to update the content of the array. So you'll probably have to copy the content from the return of  `LcdProgressBar::Draw(value)` to a char array variable that you've attached to a `LiquidLine` object. Make sure you're really getting the string from `LcdProgressBar::Draw(value)` (by serial printing it) and that you're correctly placing it in the attached char array variable.

Good luck.
